java/unittests/infodynamics/measures/continuous/gaussian/DualTotalCorrelationCalculatorGaussianTester.java

```diff
@@ -0,0 +1,133 @@
+import junit.framework.TestCase;
+
+public class DualTotalCorrelationCalculatorGaussianTester extends TestCase {
+
+  /**
+   * For two variables, DTC is equal to mutual information.
+   */
+  public void testTwoVariables() throws Exception {
+    double[][] cov = new double[][] {{1, 0.5}, {0.5, 1}};
+
+    DualTotalCorrelationCalculatorGaussian dtcCalc = new DualTotalCorrelationCalculatorGaussian();
+    dtcCalc.initialise(2);
+    dtcCalc.setCovariance(cov);
+    double dtc = dtcCalc.computeAverageLocalOfObservations();
+
+    MutualInfoCalculatorMultiVariateGaussian miCalc = new MutualInfoCalculatorMultiVariateGaussian();
+    miCalc.initialise(1,1);
+    miCalc.setCovariance(cov, 1);
+    double mi  = miCalc.computeAverageLocalOfObservations();
+
+    assertEquals(dtc, mi, 1e-6);
+  }
+
+  /**
+   * Compare against the direct calculation of DTC as a sum of entropies using the
+   * entropy calculator.
+   */
+  public void testCompareWithEntropy() throws Exception {
+    double[][] cov = new double[][] {{1, 0.4, 0.3}, {0.4, 1, 0.2}, {0.3, 0.2, 1}};
+
+    DualTotalCorrelationCalculatorGaussian dtcCalc = new DualTotalCorrelationCalculatorGaussian();
+    dtcCalc.initialise(3);
+    dtcCalc.setCovariance(cov);
+    double dtc = dtcCalc.computeAverageLocalOfObservations();
+
+    // Calculate using an entropy calculator and picking submatrices manually
+    EntropyCalculatorMultiVariateGaussian hCalc = new EntropyCalculatorMultiVariateGaussian();
+    hCalc.initialise(3);
+    hCalc.setCovariance(cov);
+    double dtc_hCalc = -2 * hCalc.computeAverageLocalOfObservations();
+
+    hCalc.initialise(2);
+    hCalc.setCovariance(MatrixUtils.selectRowsAndColumns(cov, new int[] {0,1}, new int[] {0,1}));
+    dtc_hCalc += hCalc.computeAverageLocalOfObservations();
+    hCalc.initialise(2);
+    hCalc.setCovariance(MatrixUtils.selectRowsAndColumns(cov, new int[] {0,2}, new int[] {0,2}));
+    dtc_hCalc += hCalc.computeAverageLocalOfObservations();
+    hCalc.initialise(2);
+    hCalc.setCovariance(MatrixUtils.selectRowsAndColumns(cov, new int[] {1,2}, new int[] {1,2}));
+    dtc_hCalc += hCalc.computeAverageLocalOfObservations();
+
+    assertEquals(dtc, dtc_hCalc, 1e-6);
+
+  }
+
+  /**
+   * Confirm that the local values average correctly back to the average value
+   */
+  public void testLocalsAverageCorrectly() throws Exception {
+
+    int dimensions = 4;
+    int timeSteps = 1000;
+    DualTotalCorrelationCalculatorGaussian dtcCalc = new DualTotalCorrelationCalculatorGaussian();
+    dtcCalc.initialise(dimensions);
+
+    // generate some random data
+    RandomGenerator rg = new RandomGenerator();
+    double[][] data = rg.generateNormalData(timeSteps, dimensions,
+        0, 1);
+
+    dtcCalc.setObservations(data);
+
+    double dtc = dtcCalc.computeAverageLocalOfObservations();
+    double[] dtcLocal = dtcCalc.computeLocalOfPreviousObservations();
+
+    System.out.printf("Average was %.5f\n", dtc);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.